docker/Dockerfile

```diff
@@ -1,27 +1,42 @@
 FROM python:3.9-slim
 
+# Keeps Python from generating .pyc files in the container
+ENV PYTHONDONTWRITEBYTECODE 1
+# Turns off buffering for easier container logging
+ENV PYTHONUNBUFFERED 1
+
 # Install essential tools
 RUN apt-get -y update && apt-get install -y \
     wget \
     gnupg \
     lsb-release
 
+# Install and setup poetry
+RUN pip install -U pip \
+    && apt install -y curl netcat \
+    && curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+ENV PATH="${PATH}:/root/.poetry/bin"
+
 # Install mongotools, used for getting mongo backups
 # and redis-tools, used for redis backups.
 RUN apt-get install -y mongo-tools redis-tools
 
 # Install the Postgres 12 client, needed for pg_dumpall
+# And MariaDB client for mysqldump
 RUN sh -c 'echo "deb http://apt.postgresql.org/pub/repos/apt $(lsb_release -cs)-pgdg main" > /etc/apt/sources.list.d/pgdg.list'
 RUN wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-key add -
 RUN apt-get update -y && \
-    apt-get install -y postgresql-client-12
+    apt-get install -y postgresql-client-12 mariadb-client
```

Comment:
slightly weird PR to add this in, but it's okay. Won't do any damage, and we want it anyway.

Reply:
Poetry would be the last PR in a row so it should contain all dependencies. I was thinking in advance.